src/PropertyTypes/UriPropertyType.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Spatie\IcalendarGenerator\PropertyTypes;
+
+final class UriPropertyType extends PropertyType
+{
+    /** @var string */
+    private $uri;
+
+
+    public static function create($names, string $uri): URIPropertyType
+    {
+        return new self($names, $uri);
+    }
+
+    /**
+     * URIPropertyType constructor.
+     *
+     * @param array|string $names
+     * @param uri $uri
+     */
+    public function __construct($names, string $uri)
+    {
+        parent::__construct($names);
+
+        $this->uri = $uri;
+    }
+
+    public function getValue(): string
+    {
+        return filter_var($this->uri, FILTER_VALIDATE_URL) ? $this->uri : '';
```

Comment:
I think we don't want to add an empty property if an url is not valid, this leaves us with two options:

- Throw an exception when the url is not valid, maybe this is a bit to harsh
- Not add the property when the url is not valid

I'm more a fan of that second option, so maybe we should move this check to `src/ComponentPayload.php` and not add this property when the url is not valid.

What do you think?

Reply:
IMHO it's fine to silently dismiss if it's empty (null, '') but if there's an invalid value, throwing seems like the better solution. Without throwing, it's likely the issue would go by unnoticed and be more problematic.

(but I'm not a maintainer, just a user of the lib 🙃)